--- dapl-2.0.35/dat/common/dat_strerror.c.orig	2012-01-13 19:55:59.000000000 +0100
+++ dapl-2.0.35/dat/common/dat_strerror.c	2012-06-04 16:46:08.306764381 +0200
@@ -43,6 +43,8 @@
 #include <dat2/udat.h>
 #endif				/* __UDAPL__ */
 
+#include <stdlib.h>
+
 /*********************************************************************
  *                                                                   *
  * Internal Function Declarations                                    *
