--- dapl-2.1.3/Makefile.am.orig	2014-11-21 17:55:46.000000000 +0100
+++ dapl-2.1.3/Makefile.am	2015-02-16 19:21:59.001701607 +0100
@@ -52,7 +52,6 @@
 
 sysconf_DATA = doc/dat.conf
 
-datlibdir = $(libdir)
 if DEFINE_CMA
 dapllibofadir = $(libdir)
 endif
@@ -66,7 +65,7 @@
 dapllibomcmdir = $(libdir)
 endif
 
-datlib_LTLIBRARIES = dat/udat/libdat2.la
+lib_LTLIBRARIES = dat/udat/libdat2.la
 if DEFINE_CMA
 dapllibofa_LTLIBRARIES = dapl/udapl/libdaplofa.la
 endif
@@ -163,7 +162,8 @@
 			dat/common/dat_init.c \
 			dat/common/dat_dr.c \
 			dat/common/dat_sr.c
-dat_udat_libdat2_la_LDFLAGS = -version-info 2:0:0 $(dat_version_script) -ldl 
+dat_udat_libdat2_la_LDFLAGS = -version-info 2:0:0 $(dat_version_script)
+dat_udat_libdat2_la_LIBADD = -ldl 
 
 if DEFINE_CMA
 #
@@ -278,8 +278,8 @@
         dapl/openib_cma/device.c $(XPROGRAMS)
 
 dapl_udapl_libdaplofa_la_LDFLAGS = -version-info 2:0:0 $(daplofa_version_script) \
-				   -Wl,-init,dapl_init -Wl,-fini,dapl_fini \
-				   -lpthread -libverbs -lrdmacm $(XLIBS)
+				   -Wl,-init,dapl_init -Wl,-fini,dapl_fini
+dapl_udapl_libdaplofa_la_LIBADD = dat/udat/libdat2.la -lpthread -libverbs -lrdmacm $(XLIBS)
 endif
 
 if DEFINE_SCM				
@@ -395,8 +395,8 @@
         dapl/openib_scm/device.c $(XPROGRAMS)
 
 dapl_udapl_libdaploscm_la_LDFLAGS = -version-info 2:0:0 $(daploscm_version_script) \
-                                   -Wl,-init,dapl_init -Wl,-fini,dapl_fini \
-                                   -lpthread -libverbs $(XLIBS)
+                                   -Wl,-init,dapl_init -Wl,-fini,dapl_fini
+dapl_udapl_libdaploscm_la_LIBADD = dat/udat/libdat2.la -lpthread -libverbs -lrdmacm $(XLIBS)
 endif
 
 if DEFINE_UCM                      
@@ -512,8 +512,8 @@
         dapl/openib_ucm/device.c $(XPROGRAMS)
 
 dapl_udapl_libdaploucm_la_LDFLAGS = -version-info 2:0:0 $(daploucm_version_script) \
-                                   -Wl,-init,dapl_init -Wl,-fini,dapl_fini \
-                                   -lpthread -libverbs $(XLIBS)
+                                   -Wl,-init,dapl_init -Wl,-fini,dapl_fini
+dapl_udapl_libdaploucm_la_LIBADD = dat/udat/libdat2.la -lpthread -libverbs -lrdmacm $(XLIBS)
 endif
 
 if DEFINE_MCM                      
@@ -632,8 +632,8 @@
         dapl/openib_mcm/device.c $(XPROGRAMS)
 
 dapl_udapl_libdaplomcm_la_LDFLAGS = -version-info 2:0:0 $(daplomcm_version_script) \
-                                   -Wl,-init,dapl_init -Wl,-fini,dapl_fini \
-                                   -lpthread -libverbs -lscif $(XLIBS)
+                                   -Wl,-init,dapl_init -Wl,-fini,dapl_fini
+dapl_udapl_libdaplomcm_la_LIBADD = dat/udat/libdat2.la -lpthread -libverbs -lscif $(XLIBS)
 endif
 
 libdatincludedir = $(includedir)/dat2
